Added information on recording userId for mailchimp - not to use trait.id (#4262)
segmentio/segment-docs: src/connections/destinations/catalog/mailchimp/index.md

### Recording userId

> To record a Segment `userId` in Mailchimp, you must pass the userID as a trait on your `identify()` calls. One thing to take note is not to pass 'userId' as 'id' in trait as trait.id is a reserved trait. You can pass 'userId' in a trait name with a corresponding merge field in MailChimp. Segment doesn't automatically map the userID to any Mailchimp properties.

Reviewer: ```suggestion
To record a Segment `userId` in Mailchimp, pass the user ID as a trait on your `identify()` calls. Don't pass the `userId` as a trait ID because the  `trait.id` is a reserved trait. Instead, pass the `userId` in a trait name with the corresponding merge field in Mailchimp. Segment doesn't automatically map the user ID to a Mailchimp property.
```
